Modules/cjkcodecs/multibytecodec.c

```diff
@@ -895,6 +901,82 @@ _multibytecodec_MultibyteIncrementalEncoder_encode_impl(MultibyteIncrementalEnco
     return encoder_encode_stateful(STATEFUL_ECTX(self), input, final);
 }
 
+/*[clinic input]
+_multibytecodec.MultibyteIncrementalEncoder.getstate
+[clinic start generated code]*/
+
+static PyObject *
+_multibytecodec_MultibyteIncrementalEncoder_getstate_impl(MultibyteIncrementalEncoderObject *self)
+/*[clinic end generated code: output=9794a5ace70d7048 input=4a2a82874ffa40bb]*/
+{
+    const char *pendingbuffer = NULL;
+    Py_ssize_t pendingsize;
+    MultibyteIncrementalEncoderState state;
```

Comment:
Struct layout is different by compiler.
To build portable integer, you need to use just bytearray:

```
unsigned char state[1 + MAXPENDING*4 + 8];
...
state[0] = peindingsize;
memcpy(state+1, pendingbuffer, pendingsize);
int pos = 1 + pendingsize;
memcpy(state+pos, self->state, 8);
```

Reply:
Thanks, @methane. state should now be independent of x32/64, endianness, and compiler.

Latest commit changes:
- Replace `MultibyteIncrementalEncoderState` with byte array
- Update `test_multibytecodec` to match new byte array format (difference: `pendingsize` moved to the first byte)

Reply:
Also added `test_getstate_returns_expected_value` to `test_multibytecodec` so that the consistency of the `state` byte array structure is validated by the test suite.